File: mergeSort/mergeSort.java
public class MergeSort {

 /*
  * START DEBUGGING THE TWO METHODS BELOW
  */
  static void merge(int[] arr, int[] left, int left_size, int[] right, int right_size) {
    int i = 0, j = 0, k = 0;
    
    while (i <= left_size || j <= right_size) {
        arr[k++] = Math.min(left[j++], right[i++]);
    
    while (i < left_size) {
        arr[k++] = left[i++];
    }
    
    while (j < right_size) {
        arr[k++] = right[j++];
    }
}

  static void merge_sort(int[] arr, int size) {
    if (size < 2) {
        return;
    }
    
    int mid = size / 2;
    int[] left = new int[mid];
    int[] right = new int[size - mid];
    
    for (int i = 0; i < mid; i++) {
        left[i] = arr[i];
    }
    
    for (int i = mid; i < size; i++) {
        right[i - mid] = arr[i];
    }
    
    merge_sort(left, mid);
    merge_sort(right, size - mid);
    
    merge(arr, left, mid, right, size - mid);
}


  /*
   * driver code
   * DO NOT CHANGE THE LINES BELOW
   */
  public static void main(String[] args) {
    int[] arr = {10, 7, 8, 9, 1, 5};
    int size = arr.length;
    
    System.out.println("Before sorting:");
    for (int i = 0; i < size; i++) {
        System.out.print(arr[i] + " ");
    }
    
    merge_sort(arr, size);
    
    System.out.println("\nAfter sorting:");
    for (int i = 0; i < size; i++) {
        System.out.print(arr[i] + " ");
    }  
  }
}

File: mergeSort/mergeSort.py
# START DEBUGGING THE METHOD BELOW

def merge_sort(arr):
    if len(arr) > 1:
        mid = len(arr) // 2
        left_half = arr[:mid]
        right_half = arr[mid:]

        merge_sort(left_half)
        merge_sort(right_half)

        i = j = k = 0

        while i <= len(left_half) or j <= len(right_half):
            arr[k] = min(left_half[j], right_half[i])
            i += 1
            j += 1
            k += 1

        while i < len(left_half):
            arr[k] = left_half[i]
            i += 1
            k += 1

        while j < len(right_half):
            arr[k] = right_half[j]
            j += 1
            k += 1

    return arr

# driver code
# DO NOT CHANGE THE LINES BELOW

arr = [10, 7, 8, 9, 1, 5]

print(f"Before sorting:\n{arr}\nAfter sorting:\n{merge_sort(arr)}")
